### PR TITLE
Added support for pause/resume patching of the file system modules

### PR DESCRIPTION
**Pull request by mrbean-bremen**
- can be called from the unit test, the patcher or the fs
- see #448

### CODE REVIEW

**Comment by mrbean-bremen** [comment]
@jmcgeheeiv - please have a look!

File: pyfakefs/tests/pytest/pytest_plugin_test.py [hunk 1]
```diff
@@ -1,7 +1,24 @@
...
+    fs.pause()
+    assert fs.exists(fake_temp_file.name)
+    assert not os.path.exists(fake_temp_file.name)
+    real_temp_file = tempfile.NamedTemporaryFile()
+    assert not fs.exists(real_temp_file.name)
+    assert os.path.exists(real_temp_file.name)
+    fs.resume()
+    assert not os.path.exists(real_temp_file.name)
+    assert os.path.exists(fake_temp_file.name)
+
```
Comment (jmcgeheeiv) [inline pyfakefs/tests/pytest/pytest_plugin_test.py]: Do `resume()` again, expect an exception.

File: pyfakefs/tests/pytest/pytest_plugin_test.py [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
+def test_pause_resume(fs):
+    fake_temp_file = tempfile.NamedTemporaryFile()
+    assert fs.exists(fake_temp_file.name)
+    assert os.path.exists(fake_temp_file.name)
+    fs.pause()
+    assert fs.exists(fake_temp_file.name)
+    assert not os.path.exists(fake_temp_file.name)
+    real_temp_file = tempfile.NamedTemporaryFile()
+    assert not fs.exists(real_temp_file.name)
+    assert os.path.exists(real_temp_file.name)
```
Comment (jmcgeheeiv) [inline pyfakefs/tests/pytest/pytest_plugin_test.py]: Do `pause()` again, expect an exception.

File: pyfakefs/tests/pytest/pytest_plugin_test.py (hunk 1 above)
Comment (jmcgeheeiv) [reply pyfakefs/tests/pytest/pytest_plugin_test.py]: Actually I strongly agree.  The state before the invocation is irrelevant as long as it's paused after `pause()` and resumed after `resume()`.  The API should always be permissive in this way.

Comment (mrbean-bremen) [reply pyfakefs/tests/pytest/pytest_plugin_test.py]: Ok, glad that we are on the same page - though it took me an iteration to come to that.
Are there any changes you want to see?

**Pull request merged**